crates/oxc_transformer/src/options/transformer_options.rs

```diff
@@ -0,0 +1,168 @@
+
+        // TODO: turn on plugins based on the EnvOptions's targets
+        let env_options = get_options::<EnvOptions>("env", options, &mut errors, true);
+        let targets = match env_options.get_targets() {
+            Ok(t) => t,
+            Err(err) => {
+                errors.push(OxcDiagnostic::error(err.to_string()).into());
+                return Err(errors);
+            }
+        };
+
+        let react = if options.has_preset("react") {
+            get_options::<ReactOptions>("react", options, &mut errors, true)
+        } else {
+            let has_jsx_plugin = options.has_plugin("transform-react-jsx");
+            let has_jsx_development_plugin = options.has_plugin("transform-react-jsx-development");
+            let mut react_options = if has_jsx_plugin {
+                get_options::<ReactOptions>("transform-react-jsx", options, &mut errors, false)
+            } else {
+                get_options::<ReactOptions>(
+                    "transform-react-jsx-development",
+                    options,
+                    &mut errors,
+                    false,
+                )
+            };
+            react_options.development = options.has_plugin("transform-react-jsx-development");
+            react_options.jsx_plugin = has_jsx_plugin || has_jsx_development_plugin;
+            react_options.display_name_plugin = options.has_plugin("transform-react-display-name");
+            react_options.jsx_self_plugin = options.has_plugin("transform-react-jsx-self");
+            react_options.jsx_source_plugin = options.has_plugin("transform-react-jsx-source");
+            react_options
+        };
+
+        let es2015 =
+            ES2015Options::default().with_arrow_function(enable_plugin::<ArrowFunctionsOptions>(
+                "transform-arrow-functions",
+                options,
+                &env_options,
+                &targets,
+                &mut errors,
+            ));
+
+        let typescript =
+            get_options::<TypeScriptOptions>("transform-typescript", options, &mut errors, false);
+
+        let assumptions = if options.assumptions.is_null() {
+            CompilerAssumptions::default()
+        } else {
+            match serde_json::from_value::<CompilerAssumptions>(options.assumptions.clone()) {
+                Ok(value) => value,
+                Err(err) => {
+                    errors.push(OxcDiagnostic::error(err.to_string()).into());
+                    CompilerAssumptions::default()
+                }
+            }
+        };
+
+        if !errors.is_empty() {
+            return Err(errors);
+        }
+
+        Ok(Self {
+            cwd: options.cwd.clone().unwrap_or_default(),
+            assumptions,
+            typescript,
+            react,
+            es2015,
+        })
+    }
+}
+
+fn get_options<T: Default + DeserializeOwned>(
+    name: &str,
+    babel_options: &BabelOptions,
+    errors: &mut Vec<Error>,
+    is_preset: bool,
+) -> T {
```

Comment:
Are we able to remove this `T`?

Reply:
Now we'll replace it with this code. Do you mind that this needs to write more code?

```rust
let plugin_name = "transform-arrow-functions";
enable_plugin(plugin_name, options, &env_options, &targets).map(|options| {
    from_value::<ArrowFunctionsOptions>(options).unwrap_or_else(|err| {
        report_error(plugin_name, &err, false, &mut errors);
        ArrowFunctionsOptions::default()
    })
})
```